lib/src/serializer.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// serializer.dart
+// A serialization block, serializing wide input data onto a narrower channel.
+//
+// 2024 August 27
+// Author: desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+
+import 'package:meta/meta.dart';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// Serializes wide aggregated data onto a narrower serialization stream
+class Serializer extends Module {
+  /// Clk input
+  @protected
+  Logic get clk => input('clk');
+
+  /// Reset input
+  @protected
+  Logic get reset => input('reset');
+
+  /// Allow serialization onto the output stream when [enable] is true
+  @protected
+  Logic? get enable => input('enable');
+
+  /// Return the count as an output
+  Logic get count => output('count');
+
+  /// Return [done] = true when we have processed [deserialized] completely
+  Logic get done => output('done');
+
+  /// Aggregated data to serialize out
+  LogicArray get deserialized => input('deserialized') as LogicArray;
```

Comment:
should be `@protected`

Reply:
Can you specify the rules for this?  I am leaning toward leaving everything unprotected for now.